### PR TITLE
feat: add RequestError support

### LINKED ISSUE #2464: [FEAT]: Include `@octokit/request-error` in the bundle
### Describe the need

Nice to have `@octokit/request-error` as part of the bundle with re-exported `RequestError` instance for easy error handling in the typed projects.

```typescript
// import re-exported RequestError instance from @octokit/request-error
import { Octokit, RequestError } from 'octokit'

const octokit = new Octokit(...)

try {
  return await octokit.rest...({})
} catch (error) {
  if (error instanceof RequestError) {
    // handle Octokit error
  } else {
    // handle all other errors
  }
}
```

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by DariuszPorowski**
Resolves #2464

----

### Before the change?

* Lack of RequestError support

### After the change?

* Added RequestError support by re-exporting from `@octokit/request-error`

### Other information

* N/A

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features) `N/A`
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features) `N/A`
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Review by wolfy1339 — APPROVED** [review]

**Comment by wolfy1339** [comment]
I don't understand why the coverage is going down because of that export.

**Comment by DariuszPorowski** [comment]
@wolfy1339 I think figured out, my bad, missing top-level export and missing test. Now should be good :)

**Comment by wolfy1339** [comment]
Can you fix the conflicts, then we will be good to go

File: package.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "@octokit/plugin-rest-endpoint-methods": "^7.1.1",
     "@octokit/plugin-retry": "^4.1.3",
     "@octokit/plugin-throttling": "^5.2.2",
-    "@octokit/types": "^9.2.2"
+    "@octokit/types": "^9.2.2",
+    "@octokit/request-error": "^4.0.2"
```
Comment (wolfy1339) [inline package.json]: Can you drop this down to 3.x, 4.x drops Node 14 and v16 support and we haven't done so in this repo yet

Comment (DariuszPorowski) [reply package.json]: @wolfy1339 done :)

File: package-lock.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "octokit",
   "version": "0.0.0-development",
-  "lockfileVersion": 2,
```
Comment (wolfy1339) [inline package-lock.json]: Can you regenerate this so it is lockfileVersion 2
Version 3 isn't compatible with older versions of npm

**Review by wolfy1339 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (DariuszPorowski) [reply package.json]: @wolfy1339 btw... with downgraded versions, plugin-retry and this (request error) have issue with declarations. With the latest one problem does not exists.

image

Comment (wolfy1339) [reply package.json]: That shouldn't be the case. It's most likely VSCode needing a refresh, as sometimes it likes to cache errors that are no longer there.

Browsing the npm package, you can see the types are there: https://unpkg.com/browse/@octokit/request-error@3.0.3/dist-types/index.d.ts

**Review by gr2m — APPROVED** [review]
great PR 💐

**Pull request merged**